cli/cliutil/awscheck.go

```diff
@@ -0,0 +1,88 @@
+package cliutil
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/netip"
+	"time"
+
+	"golang.org/x/xerrors"
+)
+
+const awsIPRangesURL = "https://ip-ranges.amazonaws.com/ip-ranges.json"
+
+type AWSIPv4Prefix struct {
+	Prefix             string `json:"ip_prefix"`
+	Region             string `json:"region"`
+	Service            string `json:"service"`
+	NetworkBorderGroup string `json:"network_border_group"`
+}
+
+type AWSIPv6Prefix struct {
+	Prefix  string `json:"ipv6_prefix"`
+	Region  string `json:"region"`
+	Service string `json:"service"`
+}
+
+type AWSIPRanges struct {
+	SyncToken    string          `json:"syncToken"`
+	CreateDate   string          `json:"createDate"`
+	IPV4Prefixes []AWSIPv4Prefix `json:"prefixes"`
+	IPV6Prefixes []AWSIPv6Prefix `json:"ipv6_prefixes"`
+}
+
+func NewAWSIPRanges(ctx context.Context) (*AWSIPRanges, error) {
+	client := &http.Client{}
+	reqCtx, reqCancel := context.WithTimeout(ctx, 5*time.Second)
+	defer reqCancel()
+	req, _ := http.NewRequestWithContext(reqCtx, http.MethodGet, awsIPRangesURL, nil)
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		b, _ := io.ReadAll(resp.Body)
+		return nil, xerrors.Errorf("unexpected status code %d: %s", resp.StatusCode, b)
+	}
+
+	var out AWSIPRanges
+	err = json.NewDecoder(resp.Body).Decode(&out)
+	if err != nil {
+		return nil, xerrors.Errorf("json decode: %w", err)
+	}
+	return &out, nil
+}
+
+// CheckIP checks if the given IP address is an AWS IP.
+func (r *AWSIPRanges) CheckIP(ip netip.Addr) (bool, error) {
+	if ip.IsLoopback() || ip.IsLinkLocalMulticast() || ip.IsLinkLocalUnicast() || ip.IsPrivate() {
```

Comment:
We shouldn't be given these IP classes in `coder ping`, but this is in case we ever wanted to re-use this util.